### PR TITLE
Add `--load-character` argument to load a default character

### PR DESCRIPTION
**Pull request by bmoconno**
I was looking for a way to start learning the repo, saw issue #542 and thought it was a good idea.

I'm not a `.py` programmer, so do let me know if there are any issues. It's also my first time contributing to someone else's repo... so hopefully I'm following the correct procedure.

* Add a new `--load-character` argument to the `server.py`, that loads the specified character JSON if it exists.
* ~~Make a new `log_file_prefix` function in `chat.py` to ensure that persistent log files are named consistently.~~
* Add logic to `chat.py` to save logs after they have been cleared, and generate logs on character load.

### CODE REVIEW

**Comment by bmoconno** [comment]
Closing this PR while I fix merge issue caused by recent changes to characters.

**Comment by bmoconno** [comment]
Updated for current version of text-generation-webui

* Reworked logic to make it simpler and handle .yalm/.yml if they include that after the character name in the args
* Removed `log_file_prefix` function since it doesn't seem necessary anymore

File: modules/chat.py [hunk 1]
```diff
@@ -366,17 +370,21 @@ def load_character(character, name1, name2):
...
+        save_history(timestamp=False)
 
     if shared.args.cai_chat:
         return name1, name2, greeting, context, generate_chat_html(shared.history['visible'], name1, name2, shared.character)
     else:
         return name1, name2, greeting, context, shared.history['visible']
 
 def load_default_history(name1, name2):
-    load_character("None", name1, name2)
+    load_character(shared.character or "None", name1, name2)
```
Comment (bmoconno) [inline modules/chat.py:387]: This has the added benefit of remembering the last character they were using if they close/refresh the UI without killing the server.

**Comment by tensiondriven** [comment]
Love it!  Can't wait to get this merged.

**Comment by tensiondriven** [comment]
Can you squash this down to a single commit?  I think it will have a better chance of getting merged that way.

**Comment by bmoconno** [comment]
Closing this again, lots of changes recently to `load_character` method.

**Comment by lee-b** [comment]
Why was this closed?  It doesn't seem to have been merged.  Looks like you gave up on this implementation, @bmoconno?

**Comment by bmoconno** [comment]
> Why was this closed? It doesn't seem to have been merged. Looks like you gave up on this implementation, @bmoconno?

I have a new PR open #976, but it needs to be updated again. This one was closed because some major changes to the base text-generation-webui broke a lot of stuff.

Hopefully once I update that new PR with the added suggestion we can finally get this merged into the main branch.

**Pull request closed without merging**